File: corpus/fixtures/cs/sample.cs
using System;

public static class Fixture
{
    public static string Hello(string name) => "hello " + name;

    public interface IGreeter
    {
        string Greet(string name);
    }

    public class EnglishGreeter : IGreeter
    {
        public string Greet(string name) => Hello(name);
    }

    public class SpanishGreeter : IGreeter
    {
        public string Greet(string name) => "hola " + name;
    }

    public static string ChainA(string name) => ChainB(name);
    public static string ChainB(string name) => ChainC(name);
    public static string ChainC(string name) => Hello(name);

    public static string Source() => UserInput();
    public static string UserInput() => "user";
    public static void Sink(string v) { }

    public static void TaintFlow()
    {
        Sink(Source());
    }

    public static int ClonePairA(int x, int y) => x + y + 1;
    public static int ClonePairB(int x, int y) => x + y + 1;
}

File: corpus/fixtures/go/sample.go
package fixture

// Hello returns a greeting.
func Hello(name string) string {
	return "hello " + name
}

// Greeter is implemented by concrete types.
type Greeter interface {
	Greet(name string) string
}

type EnglishGreeter struct{}

func (EnglishGreeter) Greet(name string) string { return Hello(name) }

type SpanishGreeter struct{}

func (SpanishGreeter) Greet(name string) string { return "hola " + name }

// CallChain demonstrates a multi-hop call chain.
func ChainA(name string) string { return ChainB(name) }
func ChainB(name string) string { return ChainC(name) }
func ChainC(name string) string { return Hello(name) }

// Source is a synthetic taint source.
func Source() string { return userInput() }

func userInput() string { return "user" }

// Sink is a synthetic taint sink.
func Sink(v string) { _ = v }

func TaintFlow() {
	Sink(Source())
}

// ClonePairA and ClonePairB are intentionally similar for clone detection.
func ClonePairA(x, y int) int { return x + y + 1 }
func ClonePairB(x, y int) int { return x + y + 1 }
